tools/accuracy_checker/accuracy_checker/adapters/pose_estimation_3d.py

```diff
@@ -0,0 +1,254 @@
+        launcher_2d_config = OrderedDict([
+            ('type', HumanPoseAdapter.__provider__),
+            ('part_affinity_fields_out', launcher_config['part_affinity_fields_out']),
+            ('keypoints_heatmap_out', launcher_config['keypoints_heatmap_out'])])
+        self.pose_adapter = HumanPoseAdapter(launcher_2d_config, label_map, output_blob)
+
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters.update({
+            'features_3d_out': StringField(description="Name of output layer with 3d features."),
+            'keypoints_heatmap_out': StringField(description="Name of output layer with keypoints heatmaps."),
+            'part_affinity_fields_out': StringField(
+                description="Name of output layer with keypoints pairwise relations (part affinity fields)."
+            )
+        })
+
+        return parameters
+
+    def validate_config(self):
+        super().validate_config(on_extra_argument=ConfigValidator.WARN_ON_EXTRA_ARGUMENT)
+
+    def configure(self):
+        self.features_3d = self.get_value_from_config('features_3d_out')
+        self.part_affinity_fields = self.get_value_from_config('part_affinity_fields_out')
+        self.keypoints_heatmap = self.get_value_from_config('keypoints_heatmap_out')
+
+    def process(self, raw, identifiers=None, frame_meta=None):
+        result = []
+        raw_outputs = self._extract_predictions(raw, frame_meta)
+        raw_output = zip(
+            identifiers, raw_outputs[self.features_3d], raw_outputs[self.keypoints_heatmap],
+            raw_outputs[self.part_affinity_fields], frame_meta
+        )
+        for identifier, features, heatmap, paf, meta in raw_output:
+            poses_2d = self._extract_poses_2d(heatmap, paf, meta, identifier)
+            height, width, _ = meta['image_size']
+            scale_y = height / features.shape[1]
+            scale_x = width / features.shape[2]
+            panoptic_poses_3d, translations, panoptic_poses_2d = HumanPose3dAdapter._parse_poses(
+                features, poses_2d, scale_y, scale_x, 1 / scale_x)
+            frame_result = PoseEstimation3dPrediction(
+                identifier, panoptic_poses_2d[:, 0:-1:3], panoptic_poses_2d[:, 1:-1:3], panoptic_poses_2d[:, 2:-1:3],
+                panoptic_poses_2d[:, -1], x_3d_values=panoptic_poses_3d[:, 0::4],
+                y_3d_values=panoptic_poses_3d[:, 1::4], z_3d_values=panoptic_poses_3d[:, 2::4],
+                translations=translations)
+            result.append(frame_result)
+
+        return result
+
+    def _extract_poses_2d(self, heatmap, paf, meta, identifier):
+        height, width, _ = meta['image_size']
+        heatmap_avg = np.zeros((height, width, 19), dtype=np.float32)
+        paf_avg = np.zeros((height, width, 38), dtype=np.float32)
+        pad = meta.get('padding', [0, 0, 0, 0])
+        transpose_order = (1, 2, 0) if heatmap.shape[0] == 19 else (0, 1, 2)
```

Comment:
do you really need this ternary operator?
In `HumanPoseAdapter` it is used only because we support model from TF, where NHWC layout.